src/peer.rs

```diff
@@ -36,6 +36,16 @@ impl Peer {
         &self.addr
     }
 
+    /// Returns the age.
+    pub fn age(&self) -> u8 {
+        self.age
+    }
+
+    // Converts this info into one with the input age.
+    pub fn into_age(self, age: u8) -> Self {
+        Self { age, ..self }
+    }
```

Comment:
The `into_*` convention usually signifies that you are converting one type into another one. Here you are only changing the age which makes it a confusing name.  How about `change_age` or `with_age` ?